src/rules/multiplication/A.jl

```diff
@@ -52,3 +52,30 @@ end
 
     return NormalWeightedMeanPrecision(dot(tmp, μ_out), W)
 end
+
+#----- Univariate Distributions ---#
+@rule typeof(*)(:A, Marginalisation) (m_out::UnivariateGaussianDistributionsFamily, m_in::UnivariateGaussianDistributionsFamily, meta::Union{<:AbstractCorrection, Nothing}) = begin
+    μ_in, var_in = mean_var(m_in)
+    μ_out, var_out = mean_var(m_out)
+    log_backwardpass = (x) -> -log(abs(x)) - 0.5 * log(2π * (var_in + var_out / x^2)) - 1 / 2 * (μ_out - x * μ_in)^2 / (var_in * x^2 + var_out)
+    return ContinuousUnivariateLogPdf(log_backwardpass)
+end
+
+# m_in and m_out are in Univariate Distribution family 
+@rule typeof(*)(:A, Marginalisation) (m_out::UnivariateDistribution, m_in::UnivariateDistribution, meta::Union{<:AbstractCorrection, Nothing}) = begin
+    nsamples = 3000
+    samples_in = rand(m_in, nsamples)
+    p = make_inversedist_message(samples_in, m_out)
+    return ContinuousUnivariateLogPdf(p)
+end
+
+function make_inversedist_message(samples_in, d_out)
+    return let samples_in = samples_in, d_out = d_out
+        (x) -> begin
+            result = mapreduce(+, zip(samples_in)) do (samplein,)
+                return abs(samplein) * pdf(d_out, x * samplein)
+            end
+            return log(result)
+        end
+    end
+end
```

Comment:
To my taste, I just usually opt to use sum and dot broadcasting syntax, e.g.
```julia
function make_inversedist_message(samples_in, d_out)
      return let samples_in = samples_in, d_out = d_out
          (x) -> begin
              result = sum(abs.(samples_in) .* pdf(d_out, x .* samples_in))
              return log(result)
          end
      end
  end

```
Doesn't `mapreduce` allocate intermediate arrays?
Perhaps, @bvdmitri can tell whether sum and broadcasting dot is better here; benchmarks show that these implementations are equivalent.

Reply:
It is the other way around. Broadcasting usually allocates intermediate arrays and is less efficient.
But `zip(samples_in)` with a single argument certainly does not really make sense, you can simplify it and write:

```julia
result = mapreduce(+, samples_in) do samplein
    return abs(samplein) * pdf(d_out, x * samplein)
end
```

Reply:
Thanks guys! I'll fix it